From 361ddf2d69082a2f43fa58e14a407833bc62827d Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Wed, 25 Nov 2015 17:50:06 +0800
Subject: [PATCH] arm64: rockchip: rockchip_defconfig updates for 4.4

- disable MODULES because gcc-4.9 unable to build modules
- disable some ERRATUM do not affects our SoCs
- NR_CPUS=8
- KSM=y
- enable SERIAL_8250
- disable selinux on boot else Android will failed to run

Change-Id: If08d6142fb2196798a9243a2291df46c1e3a6ed8
Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 arch/arm64/configs/rockchip_defconfig | 36 ++++++++++++++++-----------
 1 file changed, 21 insertions(+), 15 deletions(-)

diff --git a/arch/arm64/configs/rockchip_defconfig b/arch/arm64/configs/rockchip_defconfig
index 526c568098ca..a8801427e974 100644
--- a/arch/arm64/configs/rockchip_defconfig
+++ b/arch/arm64/configs/rockchip_defconfig
@@ -1,5 +1,6 @@
 # CONFIG_LOCALVERSION_AUTO is not set
 CONFIG_SYSVIPC=y
+# CONFIG_CROSS_MEMORY_ATTACH is not set
 CONFIG_AUDIT=y
 CONFIG_NO_HZ=y
 CONFIG_HIGH_RES_TIMERS=y
@@ -25,16 +26,20 @@ CONFIG_EMBEDDED=y
 # CONFIG_SLUB_DEBUG is not set
 CONFIG_PROFILING=y
 CONFIG_JUMP_LABEL=y
-CONFIG_MODULES=y
-CONFIG_MODULE_UNLOAD=y
 CONFIG_PARTITION_ADVANCED=y
 CONFIG_CMDLINE_PARTITION=y
 # CONFIG_IOSCHED_DEADLINE is not set
 CONFIG_ARCH_ROCKCHIP=y
+# CONFIG_ARM64_ERRATUM_826319 is not set
+# CONFIG_ARM64_ERRATUM_827319 is not set
+# CONFIG_ARM64_ERRATUM_824069 is not set
+# CONFIG_ARM64_ERRATUM_819472 is not set
+# CONFIG_ARM64_ERRATUM_832075 is not set
+# CONFIG_CAVIUM_ERRATUM_22375 is not set
+# CONFIG_CAVIUM_ERRATUM_23154 is not set
 CONFIG_BLOCK_RKNAND=y
-CONFIG_ARM64_ERRATUM_845719=y
-CONFIG_SMP=y
 CONFIG_SCHED_MC=y
+CONFIG_NR_CPUS=8
 CONFIG_DISABLE_CPU_SCHED_DOMAIN_BALANCE=y
 CONFIG_SCHED_HMP=y
 CONFIG_HMP_FAST_CPU_MASK="4-7"
@@ -43,7 +48,7 @@ CONFIG_HMP_VARIABLE_SCALE=y
 CONFIG_HMP_FREQUENCY_INVARIANT_SCALE=y
 CONFIG_PREEMPT=y
 CONFIG_ARMV7_COMPAT=y
-# CONFIG_CROSS_MEMORY_ATTACH is not set
+CONFIG_KSM=y
 CONFIG_CMA=y
 CONFIG_ZSMALLOC=y
 CONFIG_SECCOMP=y
@@ -52,7 +57,6 @@ CONFIG_SECCOMP=y
 CONFIG_COMPAT=y
 CONFIG_PM_AUTOSLEEP=y
 CONFIG_PM_WAKELOCKS=y
-CONFIG_PM_RUNTIME=y
 CONFIG_PM_DEBUG=y
 CONFIG_WQ_POWER_EFFICIENT_DEFAULT=y
 CONFIG_SUSPEND_TIME=y
@@ -84,7 +88,6 @@ CONFIG_INET_AH=y
 CONFIG_INET_ESP=y
 CONFIG_INET_IPCOMP=y
 # CONFIG_INET_LRO is not set
-CONFIG_IPV6=y
 CONFIG_IPV6_PRIVACY=y
 CONFIG_IPV6_ROUTER_PREF=y
 CONFIG_IPV6_ROUTE_INFO=y
@@ -328,6 +331,10 @@ CONFIG_ROCKCHIP_REMOTECTL_PWM=y
 # CONFIG_SERIO is not set
 # CONFIG_VT is not set
 # CONFIG_LEGACY_PTYS is not set
+CONFIG_SERIAL_8250=y
+# CONFIG_SERIAL_8250_DEPRECATED_OPTIONS is not set
+CONFIG_SERIAL_8250_CONSOLE=y
+CONFIG_SERIAL_8250_DW=y
 CONFIG_SERIAL_ROCKCHIP=y
 CONFIG_SERIAL_ROCKCHIP_CONSOLE=y
 # CONFIG_HW_RANDOM is not set
@@ -459,7 +466,7 @@ CONFIG_HID_ICADE=y
 CONFIG_HID_TWINHAN=y
 CONFIG_HID_KENSINGTON=y
 CONFIG_HID_LCPOWER=y
-CONFIG_HID_LENOVO_TPKBD=y
+CONFIG_HID_LENOVO=y
 CONFIG_HID_LOGITECH=y
 CONFIG_HID_LOGITECH_DJ=y
 CONFIG_LOGITECH_FF=y
@@ -476,7 +483,6 @@ CONFIG_PANTHERLORD_FF=y
 CONFIG_HID_PETALYNX=y
 CONFIG_HID_PICOLCD=y
 CONFIG_HID_PRIMAX=y
-CONFIG_HID_PS3REMOTE=y
 CONFIG_HID_ROCCAT=y
 CONFIG_HID_SAITEK=y
 CONFIG_HID_SAMSUNG=y
@@ -549,7 +555,6 @@ CONFIG_RTC_DRV_RC5T619=y
 CONFIG_DMADEVICES=y
 CONFIG_PL330_DMA=y
 CONFIG_STAGING=y
-CONFIG_ANDROID=y
 CONFIG_ASHMEM=y
 CONFIG_ANDROID_LOGGER=y
 CONFIG_ANDROID_TIMED_GPIO=y
@@ -569,6 +574,7 @@ CONFIG_FIQ_DEBUGGER_CONSOLE=y
 CONFIG_FIQ_DEBUGGER_CONSOLE_DEFAULT_ENABLE=y
 CONFIG_FIQ_DEBUGGER_EL3_TO_EL1=y
 CONFIG_COMMON_CLK_DEBUG=y
+# CONFIG_COMMON_CLK_XGENE is not set
 CONFIG_MAILBOX=y
 CONFIG_ROCKCHIP_MAILBOX=y
 CONFIG_SCPI_PROTOCOL=y
@@ -583,12 +589,11 @@ CONFIG_PWM_ROCKCHIP=y
 CONFIG_RESET_CONTROLLER=y
 CONFIG_PHY_ROCKCHIP_USB=y
 CONFIG_RK_HEADSET=y
+CONFIG_ANDROID=y
 CONFIG_ANDROID_BINDER_IPC=y
 CONFIG_EXT2_FS=y
 CONFIG_EXT2_FS_XATTR=y
 CONFIG_EXT3_FS=y
-# CONFIG_EXT3_DEFAULTS_TO_ORDERED is not set
-CONFIG_EXT4_FS=y
 CONFIG_EXT4_FS_SECURITY=y
 CONFIG_FUSE_FS=y
 CONFIG_ISO9660_FS=y
@@ -610,18 +615,19 @@ CONFIG_NLS_ISO8859_1=y
 CONFIG_NLS_UTF8=y
 CONFIG_PRINTK_TIME=y
 CONFIG_PRINTK_PROCESS=y
+CONFIG_DEBUG_INFO=y
+CONFIG_DEBUG_FS=y
 CONFIG_MAGIC_SYSRQ=y
 CONFIG_LOCKUP_DETECTOR=y
 CONFIG_DEFAULT_HUNG_TASK_TIMEOUT=10
 # CONFIG_DEBUG_PREEMPT is not set
-CONFIG_DEBUG_INFO=y
-# CONFIG_EARLY_PRINTK is not set
 CONFIG_KEYS=y
 CONFIG_SECURITY=y
 CONFIG_SECURITY_NETWORK=y
 CONFIG_LSM_MMAP_MIN_ADDR=4096
 CONFIG_SECURITY_SELINUX=y
-CONFIG_CRYPTO_NULL=y
+CONFIG_SECURITY_SELINUX_BOOTPARAM=y
+CONFIG_SECURITY_SELINUX_BOOTPARAM_VALUE=0
 CONFIG_CRYPTO_SHA512=y
 CONFIG_CRYPTO_TWOFISH=y
 CONFIG_CRYPTO_ANSI_CPRNG=y
-- 
2.35.3

